pallets/loans/src/lib.rs

```diff
@@ -2026,3 +2030,64 @@ impl<T: Config> LoansRateProvider<AssetIdOf<T>> for Pallet<T> {
         None
     }
 }
+
+#[allow(unused_variables)]
+impl<T: Config> LoansOperator<AssetIdOf<T>, AccountIdOf<T>, BalanceOf<T>> for Pallet<T> {
+    fn do_mint(
+        supplier: &AccountIdOf<T>,
+        asset_id: &AssetIdOf<T>,
+        amount: BalanceOf<T>,
+    ) -> Result<(), DispatchError> {
+        Ok(())
+    }
+    fn do_borrow(
+        borrower: &AccountIdOf<T>,
+        asset_id: &AssetIdOf<T>,
+        amount: BalanceOf<T>,
+    ) -> Result<(), DispatchError> {
+        Ok(())
+    }
+    fn do_collateral_asset(
+        supplier: &AccountIdOf<T>,
+        asset_id: &AssetIdOf<T>,
+        enable: bool,
+    ) -> Result<(), DispatchError> {
+        Ok(())
+    }
+    fn do_repay_borrow(
+        borrower: &AccountIdOf<T>,
+        asset_id: &AssetIdOf<T>,
+        amount: BalanceOf<T>,
+    ) -> Result<(), DispatchError> {
+        Ok(())
+    }
+    fn do_redeem(
+        supplier: &AccountIdOf<T>,
+        asset_id: &AssetIdOf<T>,
+        amount: BalanceOf<T>,
+    ) -> Result<(), DispatchError> {
+        Ok(())
+    }
+}
+
+#[allow(unused_variables)]
+impl<T: Config> LoansMarketDataProvider<AssetIdOf<T>, BalanceOf<T>> for Pallet<T> {
+    fn get_market_info(asset_id: &AssetIdOf<T>) -> Result<MarketInfo, DispatchError> {
+        Ok(Default::default())
+    }
+    fn get_market_status(asset_id: &AssetIdOf<T>) -> Result<MarketStatus<Balance>, DispatchError> {
+        Ok(Default::default())
+    }
+}
+
+#[allow(unused_variables)]
+impl<T: Config> LoansPositionDataProvider<AssetIdOf<T>, AccountIdOf<T>, BalanceOf<T>>
+    for Pallet<T>
+{
+    fn get_current_borrow_balance(
+        asset_id: &AssetIdOf<T>,
```

Comment:
we can put borrower before asset_id like the others I guess and we can also remove & (ref) for asset_id because it implemented Copy and is cheap to copy